src/app/components/machinechart/machinechart.component.ts

```diff
@@ -41,15 +42,14 @@ export class MachinechartComponent implements OnInit {
 
 		setTimeout(() => {
 			try {
-				// @ts-ignore
-				if (this.chartData && this.chartData.length == 1 && this.chartData[0] == 'system_missing') {
+				if (this.chartData && this.chartData.Data.length == 1 && this.chartData.Error == 'system_missing') {
 					this.specificError = 'system_missing'
 					this.chartError = true
 					return
 				}
 				this.doChart(this.key, this.id, this.chartData)
-				//@ts-ignore
-				this.chartError = this.chartData.length <= 0 || this.chartData[0].data.length <= 1
+
+				this.chartError = this.chartData.Data.length <= 0 || this.chartData.Data[0].data.length <= 1
```

Comment:
Why is this only checking for the length of `this.chartData.Data[0].data`?
`this.chartData.Data` can contain more than one set of data (see for example `doCPUSystemCharts`), shouldn't they all be checked?

Reply:
As long as one data point is drawable, we will draw the chart

Reply:
Since `doCPUSystemCharts` pushes 3 sets of data that are all based on `cpuSystemTotal` this check will suffice.
If we ever have a chart that has a different handling, we might want to not only check the length of `this.chartData.Data[0].data` bu of all `this.chartData.Data`s.